### PR TITLE
Add account activation

### PR DESCRIPTION
**Pull request by phamminhphuong**
ready

### CODE REVIEW

**Comment by phamminhphuong** [comment]
ready

File: app/mailers/user_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class UserMailer < ApplicationMailer
+  def account_activation(user)
```
Comment (huuphuc93) [inline app/mailers/user_mailer.rb]: Xóa ()

File: app/mailers/user_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    @user = user
+    mail to: user.email, subject: "Account activation"
```
Comment (huuphuc93) [inline app/mailers/user_mailer.rb]: I18n

File: app/mailers/user_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  end
+
+  def password_reset
+    @greeting = "Hi"
```
Comment (huuphuc93) [inline app/mailers/user_mailer.rb]: I18n

File: app/models/user.rb [hunk 4]
```diff
@@ -24,12 +25,29 @@ def remember
...
+    return false if digest.nil?
+    BCrypt::Password.new(digest).is_password? token
   end
 
   def forget
     update remember_digest: nil
   end
+
+  def activate
+    update :activated, true
```
Comment (huuphuc93) [inline app/models/user.rb]: update activated: true, activated_at: Time.zone.now

File: config/routes.rb [hunk 5]
```diff
@@ -9,4 +9,5 @@
     post "/login", to: "sessions#create"
     delete "/logout", to: "sessions#destroy"
     resources :users
+    resources :account_activations, only: [:edit]
```
Comment (huuphuc93) [inline config/routes.rb]: %i() thay cho []

**Comment by huuphuc93** [comment]
commented

**Comment by phamminhphuong** [comment]
ready

File: app/helpers/account_activations_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module AccountActivationsHelper
```
Comment (huuphuc93) [inline app/helpers/account_activations_helper.rb]: Xóa file k dùng

File: app/models/user.rb [hunk 7]
```diff
@@ -24,12 +25,28 @@ def remember
     update remember_digest: User.digest(remember_token)
   end
 
-  def authenticated? remember_token
-    return false if remember_digest.nil?
-    BCrypt::Password.new(remember_digest).is_password? remember_token
+  def authenticated? attribute, token
+    digest = send "#{attribute}_digest"
```
Comment (huuphuc93) [inline app/models/user.rb]: method send dùng để làm gì

**Comment by huuphuc93** [comment]
commented

**Comment by phamminhphuong** [comment]
ready

**Comment by huuphuc93** [comment]
merged

**Pull request merged**